tracer/src/Datadog.Trace/Tagging/TagPropagation.cs

```diff
@@ -161,60 +226,33 @@ public static string ToHeader(TraceTagCollection tagsCollection, int maxOutgoing
 
                     // if tag contains invalid chars,
                     // set tag "_dd.propagation_error:encoding_error"...
-                    tagsCollection.SetTag(Tags.TagPropagationError, PropagationErrorTagValues.EncodingError);
+                    _tagsCollection.SetTag(Tags.TagPropagationError, PropagationErrorTagValues.EncodingError);
 
                     // ... and don't set the header
-                    StringBuilderCache.Release(sb);
-                    return string.Empty;
+                    IsValid = false;
                 }
 
-                if (sb.Length > 0)
+                if (_sb.Length > 0)
                 {
-                    sb.Append(TagPairSeparator);
+                    _sb.Append(TagPairSeparator);
                 }
 
-                sb.Append(tag.Key)
-                  .Append(KeyValueSeparator)
-                  .Append(tag.Value);
+                _sb.Append(tag.Key)
+                    .Append(KeyValueSeparator)
+                    .Append(tag.Value);
             }
 
-            if (sb.Length > maxOutgoingHeaderLength)
+            if (_sb.Length > _maxOutgoingHeaderLength)
             {
-                Log.Debug<int, int>("Outgoing tag propagation header is too long. Length: {0}, Maximum: {1}.", sb.Length, maxOutgoingHeaderLength);
+                Log.Debug<int, int>("Outgoing tag propagation header is too long. Length: {0}, Maximum: {1}.", _sb.Length, _maxOutgoingHeaderLength);
 
                 // if combined tags get too long for propagation headers,
                 // set tag "_dd.propagation_error:inject_max_size"...
-                tagsCollection.SetTag(Tags.TagPropagationError, PropagationErrorTagValues.InjectMaxSize);
+                _tagsCollection.SetTag(Tags.TagPropagationError, PropagationErrorTagValues.InjectMaxSize);
 
                 // ... and don't set the header
-                StringBuilderCache.Release(sb);
-                return string.Empty;
-            }
-        }
-
-        return StringBuilderCache.GetStringAndRelease(sb);
-    }
-
-    internal static bool IsValid(string key, string value)
-    {
-        // keys don't allow comma, space, or equals
-        foreach (char c in key)
-        {
-            if (c is < (char)32 or > (char)126 or ',' or ' ' or '=')
-            {
-                return false;
-            }
-        }
-
-        // values don't allow comma
-        foreach (char c in value)
-        {
-            if (c is < (char)32 or > (char)126 or ',')
-            {
-                return false;
+                IsTooLong = true;
```

Comment:
I think we can use `IsValid = false` here and remove `IsTooLong`.

Reply:
🤔 true.